packages/loader/container-utils/src/liveSummarySanitizer.ts

```diff
@@ -0,0 +1,194 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    ICommittedProposal,
+    ISummaryTree,
+    SummaryType,
+    ISummaryBlob,
+} from "@fluidframework/protocol-definitions";
+
+import { assert, bufferToString } from "@fluidframework/common-utils";
+
+function readBlobContent(content: ISummaryBlob["content"]): any {
+    const json =
+        typeof content === "string" ? content : bufferToString(content, "utf8");
+    return JSON.parse(json);
+}
+
+function validateKeyPresence(
+    keys: string[],
+    srcTree: ISummaryTree,
+): boolean {
+    const filtered = Object.keys(srcTree.tree).filter((i) =>
+        keys.includes(i),
+    );
+    return filtered.length === keys.length;
+}
+
+function createBlobItem(content: any): ISummaryBlob {
+    return {
+        content: JSON.stringify(content),
+        type: SummaryType.Blob,
+    };
+}
+
+function validateTree(liveSummary: ISummaryTree): boolean {
+    for (const key of Object.keys(liveSummary.tree)) {
+        const o = liveSummary.tree[key];
+        if (o.type === SummaryType.Handle) {
+            return false;
+        } else if (o.type === SummaryType.Tree && !validateTree(o)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+class LiveV1SummarySanitizer {
+    private readonly quorumKeyIdx = 0;
+    private readonly quorumValIdx = 1;
+    private readonly emptySummaryBlob: ISummaryBlob = {
+        content: JSON.stringify([]),
+        type: SummaryType.Blob,
+    };
+
+    constructor(private readonly srcSummary: ISummaryTree) {}
+
+    public get sanitizedCopy(): ISummaryTree {
+        const appTree = this.sanitizeAppTree(this.srcSummary);
+        const protocolTree = this.sanitizeProtocolTree(this.srcSummary);
+        return {
+            type: SummaryType.Tree,
+            tree: {
+                ".app": appTree,
+                ".protocol": protocolTree,
+            },
+        };
+    }
+
+    /**
+     * Sanitize app tree
+     * @param srcSummary - summary to transform
+     */
+    private sanitizeAppTree(srcSummary: ISummaryTree): ISummaryTree {
+        const keys = [".channels", ".electedSummarizer", ".metadata"];
+        assert(
+            validateKeyPresence(keys, srcSummary),
+            "Valid app tree keys should be present",
+        );
+
+        const parentTree = srcSummary.tree;
+        const meta = readBlobContent(
+            (parentTree[".metadata"] as ISummaryBlob).content,
+        );
+        assert(meta !== undefined, "Invalid app metadata");
+
+        return {
+            type: SummaryType.Tree,
+            tree: {
+                [".channels"]: parentTree[".channels"],
+                [".electedSummarizer"]: createBlobItem({
```

Comment:
I believe electedSummarizer and metadata should be undefined rather than trying to manually edit their values. The code should handle that correctly.

Reply:
Good point, I'll check if runtime handles it properly. Reference was summarized state of unattached container, and I have seen `electedSummarizer` there.

Reply:
`getSanitizedCopy` was removed. We are moving to option (3) as discussed here: https://github.com/microsoft/FluidFramework/issues/9442. As part of existing flow `OrderedClientElection` checks if electedSummarizer listed in snapshot is in quorum, and if not, it will drop it. This is what we discussed today - letting runtime/dds-es validate any transient data from snapshot before applying it.